### ISSUE TITLE
forward() arguments are dropped

### ISSUE THREAD

**Issue by j-adamczyk** [huggingface/optimum#688]
### System Info

```shell
In `ORTModelForImageClassification` model, the `.forward()` method for CPU drops any passed arguments.

def forward(
    self,
    pixel_values: torch.Tensor,
    **kwargs,
):
   # omitted GPU part
   # converts pytorch inputs into numpy inputs for onnx
   onnx_inputs = {
      "pixel_values": pixel_values.cpu().detach().numpy(),
   }

   # run inference
   outputs = self.model.run(None, onnx_inputs)
   logits = torch.from_numpy(outputs[self.model_outputs["logits"]])

   # converts output to namedtuple for pipelines post-processing
   return ImageClassifierOutput(logits=logits)
```

This means that e.g. `output_hidden_states=True` option will not work, since a) it is not passed to model b) explicitly only logits are returned. This is a major difference compared to transformers, since this makes vectorization (i.e. ignoring head output and using only embeddings) impossible to use directly. The only workaround I found is to replace the `classifier` attribute of the model with `nn.Identity`, but this makes using both embeddings and predictions at the same time impossible.
```


### Who can help?

@jin

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Run any `ORTModelForImageClassification` model, passing `output_hidden_states=True` during prediction.

### Expected behavior

All attributes passed to `forward` are properly handles, as in transformers.

**Comment by fxmarty** [comment]
Thanks, I agree it is a missing feature in Optimum.
